app/src/main/java/org/wikipedia/readinglist/ReadingListFragment.kt

```diff
@@ -0,0 +1,826 @@
+        update()
+    }
+
+    private fun rename() {
+        ReadingListBehaviorsUtil.renameReadingList(requireActivity(), readingList) {
+            update()
+            funnel.logModifyList(readingList, 0)
+        }
+    }
+
+    private fun finishActionMode() {
+        actionMode?.finish()
+    }
+
+    private fun beginMultiSelect() {
+        if (SearchActionModeCallback.`is`(actionMode)) {
+            finishActionMode()
+        }
+        if (!isTagType(actionMode)) {
+            appCompatActivity.startSupportActionMode(multiSelectActionModeCallback)
+        }
+    }
+
+    private fun toggleSelectPage(page: ReadingListPage?) {
+        page?.let {
+            it.selected(!it.selected())
+            if (selectedPageCount == 0) {
+                finishActionMode()
+            } else {
+                actionMode?.title = resources.getQuantityString(R.plurals.multi_items_selected, selectedPageCount, selectedPageCount)
+            }
+            adapter.notifyDataSetChanged()
+        }
+    }
+
+    private val selectedPageCount: Int
+        get() {
+            var selectedCount = 0
+            displayedLists.forEach {
+                if (it is ReadingListPage && it.selected()) {
+                    selectedCount++
+                }
+            }
+            return selectedCount
+        }
+
+    private fun unselectAllPages() {
+        readingList?.let {
+            it.pages().forEach { page ->
+                page.selected(false)
+            }
+            adapter.notifyDataSetChanged()
+        }
+    }
+
+    private val selectedPages: List<ReadingListPage>
+        get() {
+            val result = mutableListOf<ReadingListPage>()
+            readingList?.let {
+                displayedLists.forEach { list ->
+                    if (list is ReadingListPage && list.selected()) {
+                        result.add(list)
+                        list.selected(false)
+                    }
+                }
+            }
+            return result
+        }
+
+    private fun deleteSelectedPages() {
+        readingList?.let {
+            if (selectedPages.isNotEmpty()) {
+                ReadingListDbHelper.instance().markPagesForDeletion(it, selectedPages)
+                it.pages().removeAll(selectedPages)
+                funnel.logDeleteItem(it, 0)
+                ReadingListBehaviorsUtil.showDeletePagesUndoSnackbar(requireActivity(), it, selectedPages) { updateReadingListData() }
+                update()
+            }
+        }
+    }
+
+    private fun addSelectedPagesToList() {
+        if (selectedPages.isNotEmpty()) {
+            val titles = selectedPages.map { ReadingListPage.toPageTitle(it) }
+            bottomSheetPresenter.show(childFragmentManager,
+                    AddToReadingListDialog.newInstance(titles, InvokeSource.READING_LIST_ACTIVITY))
+            update()
+        }
+    }
+
+    private fun moveSelectedPagesToList() {
+        if (selectedPages.isNotEmpty()) {
```

Comment:
@cooltey  This is another subtle bug introduced by Kotlin conversion.  Notice that the getter for `selectedPages` actually modifies the selected state of the underlying pages. Therefore, any subsequent calls to `selectedPages` will return different results!  The proper way is to store the value of `selectedPages` once, and operate only on the returned list.

Reply:
🤦.  Thanks for checking and fixing it!